File: src/catalog/src/main/java/com/pkarakal/catalog/domain/models/Product.java
package com.pkarakal.catalog.domain.models;

import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

@Entity
@Table(name = "products")
public record Product(
        @Id()
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(columnDefinition = "bigserial")
        Long id,
        @Column  String name,
        @Column String description,
        @Column @Positive @NotNull Double price,
        @Nullable @OneToOne(cascade = CascadeType.ALL) @JoinColumn(name="image") Image image
) {}

File: src/frontend/types/product.ts
import {Image} from "@/types/image";

export type Product = {
    id: number;
    name: string;
    description: string;
    price: number;
    image: Image
}

export type Products = Array<Product>;

export type CartItem = {
    product_id: number;
    name: string
    quantity: number;
    price: number;
}

export type CartItems = CartItem[]
